pandas/_libs/tslibs/timestamps.pyx

```diff
@@ -878,6 +921,12 @@ cdef class _Timestamp(ABCTimestamp):
 
         if freq is None:
             freq = self.freq
+            warnings.warn(
+                "Timestamp.freq is deprecated and will be removed in a future "
+                "version. Pass `freq` to Timestamp.to_period explicitly",
```

Comment:
Can you be more explicit about the exact consequence? Currently the `freq` of the Timestamp is used to create the Period, and when that is removed, you will need to pass a `freq` explicitly to avoid an error.

Reply:
How about "In a future version, calling 'Timestamp.to_period()` without passing a 'freq' will raise an exception"